conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -536,7 +544,8 @@ TEST_F(
     auto metricGroupInfo = lzt::get_metric_group_info(
         device, ZET_METRIC_GROUP_SAMPLING_TYPE_FLAG_EVENT_BASED, false);
     metricGroupInfo = lzt::optimize_metric_group_info_list(metricGroupInfo);
-
+    ASSERT_GT(metricGroupInfo.size(), 0u)
```

Comment:
optimize_metric_group_info_list just reduces the size of the list, so the check for empty should be done before calling. However, still applies the same comment about doing it in the setup() of the fixture.

Reply:
same for all the cases below

Reply:
Moved ASSERT() as suggested. SetUp() changes have been implemented only for the zetMetricQueryTest fixture. @matcabral, I'm skipping the SetUp() related refactoring for other fixtures. Please let me know if that’s acceptable. Thank you.